File: Tercera Entrega/sdi3-37.Cli-EJB/src/com/sdi/presentation/admin/action/EliminarUsuarioAction.java
package com.sdi.presentation.admin.action;

import javax.naming.Context;
import javax.naming.InitialContext;

import com.sdi.business.AdminService;
import com.sdi.dto.User;

import alb.util.console.Console;
import alb.util.menu.Action;

public class EliminarUsuarioAction implements Action {

	private static final String ADMIN_SERVICE_JNDI_KEY = "sdi3-37/sdi3-37.EJB/EjbAdminService!com.sdi.business.impl.admin.RemoteAdminService";
	
	public EliminarUsuarioAction() {}

	@Override
	public void execute() throws Exception {
		Context ctx = new InitialContext();
		AdminService aService = (AdminService) ctx.lookup(ADMIN_SERVICE_JNDI_KEY);
		
		//Pedir datos
		Long userID = Console.readLong("Id del usuario que desea eliminar");
		
		//Comprobamos que exista y borramos
		try {
			User u = aService.findUserById(userID);
			aService.deepDeleteUser(u.getId());			
			
			Console.print("\tUsuario borrado correctamente");
		} catch(NullPointerException e) {
			Console.print("\tNo existe el usuario");
		}		
	}
}

File: Tercera Entrega/sdi3-37.Cli-C_Sharp/sdi3-37.Cli-C_Sharp/action/EliminarUsuarioAction.cs
using System;
using sdi3_37.Cli_C_Sharp.SOAP;

namespace sdi3_37.Cli_C_Sharp
{
    class EliminarUsuarioAction : Action
    {
        public EliminarUsuarioAction()
        {

        }

        public void execute()
        {
            Console.WriteLine("---------------Eliminar usuario---------------");            

            //Pedir datos
            Console.Write("Id del usuario que desea eliminar: ");
            var userId = Console.ReadLine();

            try
            {
                AdminService aService = new AdminServiceClient();

                user u = aService.findUserById(new findUserById(Convert.ToInt64(userId))).@return;
                
                if(u == null)
                {
                    Console.WriteLine("\tNo existe el usuario");
                }
                else
                {
                    aService.deepDeleteUser(new deepDeleteUser(Convert.ToInt64(userId)));
                    Console.WriteLine("\tUsuario borrado correctamente");
                }               

            } catch(NullReferenceException e)
            {
                Console.WriteLine("\tNo existe el usuario");
            } catch(Exception e)
            {
                Console.WriteLine("\tError grave");
            }

            Console.WriteLine("----------------------------------------------");
        }
    }
}
